### ISSUE TITLE
Cannot download from mtg-fast - SSL: CERTIFICATE_VERIFY_FAILED

### ISSUE THREAD

**Issue by 7x11x13** [MTG/mtg-jamendo-dataset#46]
Using Python 3.10.11 on Windows I get the following error trying to download the dataset from `mtg-fast`:
```
python3 scripts/download/download.py "F:\datasets\mtg-jamendo" --unpack --remove --from mtg-fast
Downloading raw_30s from MTG Fast mirror
From: https://cdn.freesound.org/mtg-jamendo/raw_30s/audio/raw_30s_audio-00.tar
To: F:\datasets\mtg-jamendo\raw_30s_audio-00.tar
Traceback (most recent call last):
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\urllib\request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\http\client.py", line 1283, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\http\client.py", line 1329, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\http\client.py", line 1278, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\http\client.py", line 1038, in _send_output
    self.send(msg)
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\http\client.py", line 976, in send
    self.connect()
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\http\client.py", line 1455, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\ssl.py", line 513, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\ssl.py", line 1071, in _create
    self.do_handshake()
  File "C:\Users\7x11x13\.pyenv\pyenv-win\versions\3.10.11\lib\ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1007)
```

I'm not sure if it's a problem with the mirror, or some combination of the Python version and the version of the dependencies...

**Comment by alastair** [comment]
Thanks for the report. I just checked, and the certificate for `https://cdn.freesound.org` was issued on `Saturday, 18 November 2023 at 23:52:50` and expires on `Friday, 16 February 2024 at 23:52:49`.

Can you double-check that your system clock is correct?

Are you able to open https://cdn.freesound.org/mtg-jamendo/raw_30s in a browser without a certificate error?

**Comment by 7x11x13** [comment]
Yes, my system clock is correct, and I'm able to open it in my browser

**Comment by 7x11x13** [comment]
I suspect the problem is with the `wget` library not being compatible with newer versions of Python, as it hasn't been updated in 8 years

**Comment by alastair** [comment]
I wonder if the issue is the old expired letsencrypt root (https://stackoverflow.com/a/77422352, https://discuss.python.org/t/ssl-certificate-verify-failed-error-on-python-3-9-6-windows-10/15545)

Unfortunately I'm not familiar with python on windows + SSL certificates, and so I'm unsure about where you should go to edit this.

The `wget` library that you mentioned just appears to wrap `urllib.urlretrieve` which is from the python standard library, so while the package itself is quite old it should still just be using standard python methods. However perhaps it makes sense for us to update this to use `requests` instead, which may resolve this issue.

Until now, you may be able to manually download the files with your browser (or any other download utility that can recursively download a url!)

**Comment by 7x11x13** [comment]
I just tried https://github.com/MTG/mtg-jamendo-dataset/pull/25, and it worked!

**Comment by alastair** [comment]
Great, thanks for checking! It looks like `requests` may do some more complex stuff around certificates which bypasses the error that you say. We'll continue to discuss this and look at getting the PR merged.

**Comment by dbogdanov** [comment]
Closing this as we merged PR #25.
